pkg/relayer/session/proof.go

```diff
@@ -258,6 +281,94 @@ func (rs *relayerSessionsManager) goProveClaims(
 		successProofs = append(successProofs, sessionTree)
 	}
 
-	failSubmitProofsSessionsCh <- failedProofs
+	failGenerateProofsSessionsCh <- failedProofs
 	proofsGeneratedCh <- successProofs
 }
+
+// isProofRequired determines whether a proof is required for the given session's
+// claim based on the current proof module governance parameters.
+// TODO_TECHDEBT: Once the on/off-chain loggers are unified, move this logic to
+// a shared helper used by both off-chain and on-chain routines.
+func (rs *relayerSessionsManager) isProofRequired(
+	ctx context.Context,
+	sessionTree relayer.SessionTree,
+	proofPathSeedBlock client.Block,
+) (isProofRequired bool, err error) {
+	logger := rs.logger.With(
+		"session_id", sessionTree.GetSessionHeader().GetSessionId(),
+		"claim_root", fmt.Sprintf("%x", sessionTree.GetClaimRoot()),
+		"supplier_operator_address", sessionTree.GetSupplierOperatorAddress().String(),
+	)
+
+	// Create the claim object and use its methods to determine if a proof is required.
+	claim := claimFromSessionTree(sessionTree)
+
+	// Get the number of compute units accumulated through the given session.
+	numClaimComputeUnits, err := claim.GetNumComputeUnits()
```

Comment:
Can you please create a helper for `getComputeUnits` that gets

1. Compute units from tree
2. Num relays in tree * service.ComputeUnitsPerRelay
3. Errors if (1) & (2) are not the same
4. Returns a single value

Reply:
Now that #745 is merged, It fixes [1] `smst.Update` to correctly update `numComputeUnits` for the tree, so it's safe to continue using `claim.GetNumComputeUnits`

[1] https://github.com/pokt-network/poktroll/blob/main/pkg/relayer/session/session.go#L436